application/core/LSHttpRequest.php

```diff
@@ -286,4 +285,32 @@ public function isRestRequest()
         ) === 1;
         return $restPath || $restRoute;
     }
+
+    /**
+     * @inheritdoc
+     * Check host with config['allowedHost'] if it set
+     */
+    public function getHostInfo($schema = '')
+    {
+        $hostInfo = parent::getHostInfo($schema);
+        self::checkIsAllowedHost($hostInfo);
+        return $hostInfo;
+    }
+
+    /**
+     * Check if an url are in allowed host (if exist)
+     * @var string $hostInfo
+     * @throw Exception
+     * @return void
+     */
+    public static function checkIsAllowedHost($hostInfo)
+    {
+        $allowedHosts = App()->getConfig('allowedHosts');
+        if (!empty($allowedHosts) && is_array($allowedHosts)) {
+            $host = parse_url($hostInfo, PHP_URL_HOST);
+            if ($host && !in_array($host, $allowedHosts)) {
```

Comment:
If no host, the validation is not done?

Reply:
No, because `'/resetpassword.php'` is always a valid host

It can not happen for 19639 issue , because `parent::getHostInfo($schema);` always return an host.

Current usage : there are always an host.

Reply:
if there is always a host, why checking it (on the if statement)?
What about throwing an exception if no host is found? Just in case

Reply:
PS : see the comment : “Check if an url are in allowed host **(if exist)**”

Reply:
I don't completely agree, but we don't need to 100% of times :) 
I have already set the code review as OK last week I think

Reply:
I think if there are no hosts here : there are no security issue. Host not set : no security issue in URL :)

We need to check if it's a valid host for Host Injection. 

No Host _then_ no injection _then_ no issue